module/zfs/zil.c

```diff
@@ -1253,6 +1267,17 @@ zil_lwb_write_done(zio_t *zio)
 	 * written out.
 	 */
 	if (zio->io_error != 0) {
+		zil_commit_waiter_t *zcw;
```

Comment:
I'm curious.. this block of code you're adding, looks similar to the block of code in the "flush done" function.. i.e. starting at line 1173.. but I see some differences, such as the fast that this block you're adding doesn't call:
```
cv_broadcast(&zcw->zcw_cv);
```
nor does it set `zcw_done`.. 

are these differences intentional? it's been awhile since I've been in this code, so I'm just curious if we should be using the same exact logic in both cases, here and in the flush function?

in this error case, do we still call the "flush done" function? I presume not, which is why this change is needed.. but please correct me if I'm wrong.